noir-projects/noir-protocol-circuits/crates/types/src/traits.nr

```diff
@@ -253,14 +253,14 @@ where
     fn deserialize(fields: [Field; O * M + 1]) -> Self {
         let mut new_bounded_vec: BoundedVec<T, M> = BoundedVec::new();
 
-        let len = fields[0] as u32;
-        let mut index = 1;
```

Comment:
Nuking the index mutable var here was not necessary but I did it to have it consistent with Serialize.